sandy/core/cov.py

```diff
@@ -496,11 +496,107 @@ def decompose(self):
         Q, R = scipy.linalg.qr(M.T)
         return R.T
 
-    def glls(self, S, Vy):
+    @classmethod
+    def get_covariance(cls, std):
+        """
+        Construct the covariance matrix from the variance vector.
+
+        Parameters
+        ----------
+        std : `pandas.Series`
+            Variance vector.
+
+        Returns
+        -------
+        `CategoryCov`
+            Object containing the covariance matrix.
+
+        Example
+        -------
+        >>> S = pd.Series(np.array([0, 2, 3]), index=pd.Index([1, 2, 3]))
+        >>> cov = sandy.CategoryCov.get_covariance(S)
+        >>> cov
+                    1           2           3
+        1 0.00000e+00 0.00000e+00 0.00000e+00
+        2 0.00000e+00 2.00000e+00 0.00000e+00
+        3 0.00000e+00 0.00000e+00 3.00000e+00
+
+        >>> assert type(cov) is sandy.CategoryCov
+        """
+        cov = pd.DataFrame(np.diag(std.values),

```

Comment:
maybe it is best to assume that std can be an iterable...not specifically a pandas series.

then we should check: if we have std being a series and passing to `np.diag(std)`, would it still work?
If it works, we don't have to restrict ourselves to a pandas series